### PR TITLE
Fixes/do routing the new way

### PR DESCRIPTION
**Pull request by iphands**
This is a [wip] at the moment. Its a large re-sync to the starter apps cloud routing config.
It should eliminate some of the /insights/undefined/actions and /insights/actions/rules bugs... but it needs more thorough review.

Please have a look.

### CODE REVIEW

**Comment by AllenBW** [comment]
@iphands trying to reconcile the addition of `./Profiles/*` and removal of  `config/spandx.config.js`... are these things unrelated? (couldn't find any command to use the profiles in insights proxy) how else is insights-proxy to know the paths?

File: src/App.js [hunk 1]
```diff
@@ -1,30 +1,28 @@
 import PropTypes from 'prop-types';
 import React, { Component } from 'react';
-import { routerParams } from '@red-hat-insights/insights-frontend-components';
```
Comment (karelhala) [inline src/App.js]: I think that we want to use routerParams here to store current URL and show correct breadcrumbs right?

File: src/App.js [hunk 2]
```diff
@@ -1,30 +1,28 @@
...
 class App extends Component {
     componentDidMount () {
         insights.chrome.init();
         insights.chrome.identifyApp('insights');
-        this.appNav = insights.chrome.on('APP_NAVIGATION', event => {
-            if (!matchPath(location.href, { path: `${document.baseURI}insights/${event.navId}` })) {
-                this.props.history.push(`/${event.navId}`);
-            }
-        });
+        insights.chrome.navigation(buildNavigation());
```
Comment (karelhala) [inline src/App.js]: BuildNav should not be used anymore, it was to add custom secondary level.

File: src/Routes.js [hunk 3]
```diff
@@ -52,30 +53,17 @@ function checkPaths (routes, app) {
...
 
     return (
         <Switch>
-            <InsightsRoute path={ paths.actions } component={ Actions } rootClass='actions'/>
-            <InsightsRoute path={ paths.rules } component={ Rules } rootClass='rules'/>
+            <InsightsRoute path={ pathsMap.samplepage } component={ Actions } rootClass='actions'/>
+            <InsightsRoute path={ pathsMap.rules } component={ Rules } rootClass='rules'/>
+
+            { /* Finally, catch all unmatched routes */ }
+            <Route render={ () => some(paths, p => p === path) ? null : (<Redirect to={ paths.actions }/>) }/>
```
Comment (karelhala) [inline src/Routes.js]: This should work as replacement for `checkPaths` it needs to be tested to see if bookmarking still works.

File: src/Routes.js [hunk 4]
```diff
@@ -52,30 +53,17 @@ function checkPaths (routes, app) {
...
-    }
+export const Routes = (props: Props) => {
+    const path = props.childProps.location.pathname;
+    const paths = Object.values(pathsMap);
 
     return (
         <Switch>
-            <InsightsRoute path={ paths.actions } component={ Actions } rootClass='actions'/>
-            <InsightsRoute path={ paths.rules } component={ Rules } rootClass='rules'/>
+            <InsightsRoute path={ pathsMap.samplepage } component={ Actions } rootClass='actions'/>
```
Comment (karelhala) [inline src/Routes.js]: `pathsMap.samplepage` -> `pathsMap.actions`

**Comment by AllenBW** [comment]
Well its been 7 days.. gonna close this for now, feel free to reopen anew 🤗

**Pull request closed without merging**